### DISCUSSION TITLE
Models.yml File Disappears When Uploading Models from Companion to TX16S

### DISCUSSION THREAD

**Discussion by wildmancp** [EdgeTX/edgetx#4174] [Q&A]
Hello and Good Day! I am new to RC and am stuck on a weird problem. I have always had EdgeTX and normally configure new models on the radio, but I recently started using EdgeTX Companion 2.9.1 to help setup an OMP M1. I can sync the models FROM the radio TO Companion, but I have never been able to upload my models from Companion to the radio. I found that when I push the models from Companion the individual model yml files update on the radio, but the models.yml file disappears. The resulting behavior is that the radio shows no models at all, but when you try to sync back to Companion you still see all the models because the yml files are all there. I have to manually drop the models.yml file into the MODELS folder (on the SD card) to get it working again. 

The only theory I have is that something is out of sync with the models files. For some reason on the radio there are only 17 modelX.yml files (1-17). When I push from Companion it creates a new one, model18.yml, and you see on the SD card that the models.yml file disappears and the new model18.yml appears. In the screen shots below you can see the model18.yml file in both screenshots as I added it to the radio as a troubleshooting step. No luck! I have no idea what I am doing wrong, maybe there is a file path that isn't set correctly, or something internal on the radio?

I would not have made it this far without all the posts and responses from the community! Thank you all! I hope I can one day pay it forward. 

I have updated everything, tried many different ways, but no luck
Screenshot 1 is to show everything is up to date
Screenshot 2 is a sync from the Radio to Companion. You can see all the files on the SD card on the right
Screenshot 3 is immediately after a push from Companion to the radio. You can see the models.yml file is gone 

![image1](https://github.com/EdgeTX/edgetx/assets/147096506/82bc2527-c104-4e9f-a477-2e0b2b318e34)
![image2](https://github.com/EdgeTX/edgetx/assets/147096506/c8a22152-7c19-428d-a593-bd65cda23f43)
![image3](https://github.com/EdgeTX/edgetx/assets/147096506/682b666a-f5c1-48cc-be02-e72ecba33f4d)

**Comment by gagarinlg** [comment]
The models.yml file is deprecated. Please use the companion with the same version number as EdgeTX has on your radio, then it will manage the models.yml file, or update your radio to 2.9.1

**Reply by wildmancp** [reply]
Thank you, I feel so dumb! I thought the radio was updated, but when I checked it was still on the 2.7 version. Thank you so much!

**Comment by wildmancp** [comment]
The issue was that the radio was on 2.7, but the Companion was 2.9. Thank you!
